adapter/internal/discovery/xds/server.go

```diff
@@ -272,7 +280,7 @@ func UpdateAPI(vHost string, apiProject mgw.ProjectAPI, environments []string) (
 		mgwSwagger.SetXWso2AuthHeader(apiYaml.AuthorizationHeader)
 
 	} else if apiProject.APIType == mgw.WS {
-		mgwSwagger, err = operator.GetMgwSwaggerWebSocket(apiProject.APIYaml)
+		mgwSwagger, err = operator.GetMgwSwaggerWebSocket(apiProject.APIYaml, apiEnvProps)
```

Comment:
how about adding `mgwSwagger.SetEnvProperties(envProp)` after finishing both checks at 268 and 282? maybe after 293?

Reply:
yes. that's way better. thanks